src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -529,4 +529,81 @@
       it { is_expected.to redirect_to(:back) }
     end
   end
+
+  describe 'GET #rebuild_time' do
+    before do
+      # To not ask backend for build status
+      Project.any_instance.stubs(:number_of_build_problems).returns(0)
+    end
+
+    context 'with an invalid scheduler' do
+      before do
+        get :rebuild_time, project: user.home_project, repository: repo_for_user_home.name, arch: 'x86_64', scheduler: 'invalid_scheduler'
+      end
+
+      it { expect(flash[:error]).to eq('Invalid scheduler type, check mkdiststats docu - aehm, source') }
+      it { is_expected.to redirect_to(project_show_path(user.home_project)) }
+    end
+
+    context 'without build dependency info or jobs history' do
+      before do
+        BuilddepInfo.stubs(:find).returns(nil)
+        Jobhistory.stubs(:find).returns(nil)
+        get :rebuild_time, project: user.home_project, repository: repo_for_user_home.name, arch: 'x86_64'
+      end
+
+      it { expect(flash[:error]).to start_with('Could not collect infos about repository') }
+      it { is_expected.to redirect_to(project_show_path(user.home_project)) }
+    end
+
+    context 'normal flow' do
+      before do
+        BuilddepInfo.stubs(:find).returns([])
+        Jobhistory.stubs(:find).returns([])
+      end
+
+      context 'with diststats generated' do
+        before do
+          path = Xmlhash::XMLHash.new({'package' => 'package_name' })
+          longestpaths_xml = Xmlhash::XMLHash.new({ 'longestpath' => Xmlhash::XMLHash.new({'path' => path }) })
+          Webui::ProjectController.any_instance.stubs(:call_diststats).returns(longestpaths_xml)
+          get :rebuild_time, project: user.home_project, repository: repo_for_user_home.name, arch: 'x86_64'
+        end
+
+        it { expect(assigns(:longestpaths)).to match_array([[], [], [], [], ["package_name"]]) }
+      end
+
+      context 'with diststats not generated' do
+        before do
+          Webui::ProjectController.any_instance.stubs(:call_diststats).returns(nil)
+          get :rebuild_time, project: user.home_project, repository: repo_for_user_home.name, arch: 'x86_64'
+        end
+
+        it { expect(assigns(:longestpaths)).to match_array([[], [], [], []]) }
+      end
+    end
+  end
+
+  describe 'GET #rebuild_time_png' do
```

Comment:
Cache data writing is not in the scope of this method (rebuild_time_png), isn't it? I tested that it reads from the cache using a key in the way it accesses.. and that the content in the cache is sent properly and with the headers too. This action doesn't do anything more...

What should I test? Cache data writing is performed in ```call_diststats``` method that is called in ```rebuild_time``` action.